models/biocro/R/met2model.BIOCRO.R

```diff
@@ -16,23 +16,60 @@
 ##' @title Write BioCro met files
 ##' @param in.path path on disk where CF file lives
 ##' @param in.prefix prefix for each file
-##' @param outfolder location where model specific output is written.
-##' @param ... can pass lat, lon, start.date and end.date
-##' @return OK if everything was succesful.
+##' @param outfolder location where model specific output is written
+##' @param lat, lon Site latitude and longitude
+##' @param start_date, end_date Date range to convert. Each year will be written to a separate file
+##' @param ... other arguments passed from PEcAn, currently ignored
+##' @return invisibly, a dataframe of information about the written file.
 ##' @export
 ##' @author Rob Kooper, David LeBauer
 ##-------------------------------------------------------------------------------------------------#
-met2model.BIOCRO <- function(in.path, in.prefix, outfolder, overwrite = FALSE, ...) {
-  ncfiles <- dir(in.path, full.names = TRUE, pattern = paste0(in.prefix, "*.nc$"), 
-                 all.files = FALSE, recursive = FALSE)
-  metlist <- list()
-  for (file in ncfiles) {
-    met.nc <- ncdf4::nc_open(file)
-    tmp.met <- load.cfmet(met.nc, lat = lat, lon = lon, start.date = start.date, 
-                          end.date = end.date)
-    metlist[[file]] <- cf2biocro(tmp.met)
+met2model.BIOCRO <- function(in.path, in.prefix, outfolder, overwrite = FALSE,
+                             lat, lon, start_date, end_date, ...) {
+  library(PEcAn.all)
+  dir.create(file.path(outfolder), recursive = TRUE, showWarnings = FALSE)
+  years_wanted <- lubridate::year(start_date):lubridate::year(end_date)
+  out.prefix <- "BIOCRO_MET_INPUT"
```

Comment:
Looking at the SIPNET workflow -> 
  /home/carya/output/dbfiles/AmerifluxLBL_CF_gapfill_site_0-622/AMF_US-NR1_BASE_HH_9-1.2004.nc gets converted in met2model to
  /home/carya/output/dbfiles/AmerifluxLBL_SIPNET_site_0-772/AMF_US-NR1_BASE_HH_9-1.2004-01-01.2004-12-31.clim

In this case, the directory name keeps the formatname_modelname_siteID/, the basename passed from the file is forwarded through, and then start date and end date are appended, and then the extension is matching what the model expects (sipnet expects .clim ASCII files)